### PR TITLE
settings via java objects

### PR DESCRIPTION
**Pull request by l3r8yJ**
@h1alexbel take a look, please 

say what you think, if it's okay i'll write javadocs, tests and etc

### CODE REVIEW

**Comment by l3r8yJ** [comment]
@h1alexbel I also have an idea where you can point to the class directly, ie:
```java
new KeyDeserializer(StringDeserializer.class);
```

**Review by h1alexbel — APPROVED** [review]
wow! @l3r8yJ cool

**Comment by h1alexbel** [comment]
> @h1alexbel I also have an idea where you can point to the class directly, ie:
> 
> ```java
> new KeyDeserializer(StringDeserializer.class);
> ```
> 
> But I'm not sure that's a good idea.

we can do something like this:
```java
ser = new LongSerializer();
new KeyDeserializer(ser);
```

**Comment by h1alexbel** [comment]
@l3r8yJ will be awesome if you atach javadocs to new classes

**Comment by l3r8yJ** [comment]
@h1alexbel yes, i'll write them. 
just waiting for approval, there's just no point in writing tests/java docs to delete everything later) 
I'll send an update later to this pr

File: src/main/java/io/github/eocqrs/kafka/settings/KfParams.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+/**
+ * It's a collection of `{@link SettingsAttribute}` objects.
+ *
+ * @author Ivan Ivanchuk (l3r8y@duck.com)
+ * @since 0.0.2
+ */
+public final class KfParams implements Settings {
```
Comment (h1alexbel) [inline src/main/java/io/github/eocqrs/kafka/settings/KfParams.java:39]: @l3r8yJ I have an idea: rename Settings interface to `Params`
we already have ConsumerSettings and ProducerSettings.
WDYT?

Comment (l3r8yJ) [reply src/main/java/io/github/eocqrs/kafka/settings/KfParams.java:39]: @h1alexbel good idea

**Comment by l3r8yJ** [comment]
@h1alexbel take a look, please

File: src/test/java/io/github/eocqrs/kafka/producer/settings/KfProducerParamsTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ */
+final class KfProducerParamsTest {
+
+  @Test
+  void representsXmlCorrectly() {
+    MatcherAssert.assertThat(
+      "Represents right XML settings",
+      new KfProducerParams(
+        new KfParams(
+          new GroupId("103")
```
Comment (h1alexbel) [inline src/test/java/io/github/eocqrs/kafka/producer/settings/KfProducerParamsTest.java:45]: GroupId can be attached only to consumer

Comment (l3r8yJ) [reply src/test/java/io/github/eocqrs/kafka/producer/settings/KfProducerParamsTest.java:45]: @h1alexbel this is only for the reproduced xml check, no more, need to be fixed?

Comment (h1alexbel) [reply src/test/java/io/github/eocqrs/kafka/producer/settings/KfProducerParamsTest.java:45]: @l3r8yJ no, just heads up

**Comment by h1alexbel** [comment]
@rultor merge

**Pull request merged**